src/Field/Definition/AssetField.php

```diff
@@ -65,21 +65,35 @@ public function toManagementApiData (int $position) : array
 	 */
 	public function validateData (ComponentContext $context, array $contentPath, mixed $data) : void
 	{
+		$idConstraints = [
+			new Type("string"),
+		];
+		$fileNameConstraints = [
+			new Type("string"),
+		];
+		$isExternalUrlConstraints = [
+			new Type("bool"),
+		];
```

Comment:
The original implementation already had a `NotNull` constraint. I wasn't sure whether this was a bug or if it's intentional — especially since it's in the „optional fields” section.

Reply:
When the field is provided, then it was always not-null.

Reply:
So should we always check for `NotNull`?

Reply:
Not sure, that was just to answer your question. I think it's fine like it is now. 👍